src/irmin-pack/inode_intf.ml

```diff
@@ -19,6 +19,59 @@ module type Val_intf = sig
 
   val pred : t -> [ `Node of hash | `Inode of hash | `Contents of hash ] list
 
+  module type DATA_FORMAT = sig
+    type vt
+    type t
+
+    exception Misconstructed_Data of string
```

Comment:
```suggestion
    exception Misconstructed_data of string
```

I think `RO_Not_Allowed` is an odd one out here; let's not follow it.

Reply:
What is this `RO_Not_Allowed` thing?

Reply:
AFAIK, the only other exception in the codebase that uses this naming convention.

Reply:
Ok ;-) Is there a document for Irmin conventions so you don't have to fix these kind of mistakes?

Reply:
I'm not aware of one, but perhaps now is the time to write it :-)